pkg/metrics/exporters/stackdriver/stackdriver.go

```diff
@@ -31,26 +41,33 @@ func Start(ctx context.Context) error {
 		return err
 	}
 
-	exporter, err := stackdriver.NewExporter(stackdriver.Options{
-		MetricPrefix:      metricPrefix,
-		MonitoredResource: monitoredresource.Autodetect(),
-	})
+	e, err := stackdriver.New(stackdriver.WithMetricDescriptorTypeFormatter(func(desc metricdata.Metrics) string {
+		return fmt.Sprintf("%s/%s", metricPrefix, desc.Name)
+	}))
 	if err != nil {
 		if *ignoreMissingCreds {
 			log.Error(err, "Error initializing stackdriver exporter, not exporting stackdriver metrics")
 			return nil
 		}
 		return err
 	}
-
-	if err := exporter.StartMetricsExporter(); err != nil {
-		if *ignoreMissingCreds {
-			log.Error(err, "Error starting stackdriver exporter, not exporting stackdriver metrics")
-			return nil
-		}
+	res, err := resource.New(ctx,
+		resource.WithDetectors(gcp.NewDetector(), ec2.NewResourceDetector()),
+		resource.WithTelemetrySDK(),
+		resource.WithFromEnv(),
+	)
+	if err != nil {
 		return err
 	}
-	defer exporter.StopMetricsExporter()
+	reader := metric.NewPeriodicReader(e, metric.WithInterval(*metricInterval))
+	meterProvider := metric.NewMeterProvider(
+		metric.WithReader(reader),
+		metric.WithView(view.Views()...),
+		metric.WithResource(res),
+	)
+
+	otel.SetMeterProvider(meterProvider)
```

Comment:
Another note is that I don't think this current setup works for multiple `exporters` although I don't have a good a insight as the comment above. Again, I think there's a race for which exporter runnable is started last. For instance, when I configure `stackdriver, prometheus` only prometheus metrics show up.

Reply:
Implemented common lib for setting global `meterprovider` with expected readers. It should fix the issue with running multiple `exporters`.